src/Symfony/Component/Security/Core/Tests/Authorization/Voter/CallableVoterTest.php

```diff
@@ -0,0 +1,135 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Security\Core\Tests\Authorization\Voter;
+
+use PHPUnit\Framework\TestCase;
+use Symfony\Component\Security\Core\Authorization\Voter\CallableVoter;
+use Symfony\Component\Security\Core\Authorization\Voter\Voter;
+
+class CallableVoterTest extends TestCase
+{
+    /**
+     * @dataProvider getVoteTests
+     */
+    public function testVote($callable, $expected)
+    {
+        $voter = new CallableVoter($callable);
+        $tokenMock = $this->getMockBuilder('Symfony\Component\Security\Core\Authentication\Token\TokenInterface')->getMock();
+
+        $this->assertSame($expected, $voter->vote($tokenMock, null, array()));
+    }
+
+    public function getVoteTests()
+    {
+        return array(
+            array(
+                function () {
+                    return true;
+                },
+                Voter::ACCESS_GRANTED,
+            ),
+            array(
+                \Closure::fromCallable(function () {
```

Comment:
why wrapping in `Closure::fromCallable`?

Reply:
to test with `\Closure` object.
You think it's useless?

Reply:
It's useless then, yes: a closure is already a `Closure`

Reply:
You got me 😆...